New yaml config branch

allenmichael099 opened the pull request:
This branch is for work on the YAML config file. I added a new argument to the run_app function called 'config_file', which is the path and file name for the config file on your machine. Mine is ~emdash/R/config.yml.

From my most recent commit:
- YAML file made to configure options for the dashboard.
- It gets loaded using config::get within run_app, and an emdash.varName is set for each variable in the YAML
- The app_ui checks display signup trend, and the app_server uses cols_to_remove_from_particpts_table and col_labels_for_participts.
- Added a few more labels that may appear in Shankari's last PR, and added the skip_absent = TRUE argument to setnames in app_server
- Added a map to the yml for the participant table labels: col_labels_for_participts

allenmichael099 committed:
column names and tag font size

In app_server:
Changed the displayed column names so that they are easier to understand.
Used setnames within observeEvent for the esquisse plots and for the data table.

In app_ui:
changed the font size for the variable tags within box() after esquisserUI was called

allenmichael099 committed:
Working on YAML config file

YAML file made to configure options for the dashboard.

It gets loaded using config::get within run_app, and an emdash.<varName> is set for each variable in the YAML

The app_ui checks display signup trend, and the app_server uses cols_to_remove_from_particpts table.

I added a few more labels that may appear in Shankari's last PR, and added the skip_absent = TRUE argument to setnames in app_server

allenmichael099 committed:
Revert "Working on YAML config file"

This reverts commit c94773c7b241d80d42145275bf0a1970c3739b11.

allenmichael099 committed:
Revert "Revert "Working on YAML config file""

This reverts commit e7fd77c7bb758644654afcad69badef2aab58eb8.

allenmichael099 committed:
Added to a YAML config file

- YAML file made to configure options for the dashboard.
- It gets loaded using config::get within run_app, and an emdash.<varName> is set for each variable in the YAML
- The app_ui checks display signup trend, and the app_server uses cols_to_remove_from_particpts_table and col_labels_for_participts.
- Added a few more labels that may appear in Shankari's last PR, and added the skip_absent = TRUE argument to setnames in app_server
- Added a map to the yml for the participant table labels: col_labels_for_participts

allenmichael099 commented:
Here are images of some of the configurable features. The pictures on the left correspond to the left yaml script. The right yaml script is altered to show the sign up trend and to not change the user_id and update_ts labels.
![image](https://user-images.githubusercontent.com/56986329/120560716-76406180-c3c0-11eb-8260-e41dcef47252.png)
![image](https://user-images.githubusercontent.com/56986329/120560408-e4d0ef80-c3bf-11eb-8415-d8da0a09cfb2.png)

allenmichael099 committed:
Change label size

changed label size from 12 to 14

asiripanich commented on R/run_app.R:
```diff
@@ -13,14 +13,22 @@
 #' @importFrom shiny shinyApp
 #' @importFrom golem with_golem_options
-run_app <- function(mongo_url, anon_locations = FALSE, ...
+run_app <- function(config_file,mongo_url, anon_locations = FALSE, ...
```
Let's not alter the argument order. How about adding the `config_file` argument after `mongo_url` and move `anon_locations` to the config file?

allenmichael099 replied:
@asiripanich I switched the order to what you suggested and moved anon_locations into the config. I was also thinking I would move mongo_url into the config as well, if that's okay with you.

asiripanich replied:
Yes, let's move `mongo_url` too.

asiripanich commented on R/app_ui.R:
```diff
@@ -37,18 +38,22 @@ app_ui <- function(request) {
                   # Dashboard - plots - start ----------------------------------
+                  
+                  # Display the signup trend if that option is chosen  
+                  if (getOption("emdash.disp_signup_trend")){
```
To be safe, let's check for `TRUE`. You can wrap the value around `isTRUE(...)`.

asiripanich commented on R/app_server.R:
```diff
@@ -80,11 +80,19 @@ app_server <- function( input, output, session ) {
         data_r$participants %>%
-        drop_list_columns()
+        drop_list_columns() %>%
+        setnames(originalColumnNames,newColumnNames,skip_absent = TRUE)
```
I also think this is a good idea. The best practice is to use `data.table::setnames(...)` instead of commenting if you think it is necessary to note.

asiripanich commented on R/app_server.R:
```diff
@@ -106,10 +114,11 @@ app_server <- function( input, output, session ) {
-  cols_to_remove_from_participts_table <- c("first_trip_datetime", 
-                                              "last_trip_datetime")
+  cols_to_remove_from_participts_table <- getOption('emdash.cols_to_remove_from_participts_table')
   cols_to_remove_from_trips_table <- c("start_fmt_time0", "start_local_dt_timezone", "start_fmt_time",
```
For consistency, let's move this to the config file as well.

allenmichael099 replied:
Got it, I moved the 'cols to remove' from both trips and map_popup into the config.

asiripanich commented:
Great work @allenmichael099! I think we should put a default config file in the `inst` directory, so that people can run the dashboard without having to create a config file themselves, what do you think?

allenmichael099 commented:
Thanks @asiripanich! That sounds good to me. We may want to go over what the defaults should be.

asiripanich reviewed (commented):
Here are my comments.

allenmichael099 committed:
YAML Additions

- Moved anon_locations and cols_to_remove_from_ (trips and map_popup) to the YAML
- wrote 'data.table::' in front of setnames in the server
- changed the anon_locations and disp_signup_trend checks to use ifTRUE()

allenmichael099 committed:
Update run_app.R

Moved checkmate::assert_flag to check config$anon_locations instead of anon_locations

asiripanich commented:
Looking good, @allenmichael099! I will review this again after our meeting :).

asiripanich commented on R/config.yml:
```diff
@@ -1,10 +1,7 @@
-# "first_diary_call", "last_diary_call")
+# Use this file to set options for the app
 
 default:
```
This file should be stored in `inst` folder.

asiripanich replied:
Fixes in https://github.com/asiripanich/emdash/pull/32/commits/c9aeb173bc098669dc37c82d0f03d412780b6e3c

asiripanich committed:
Merge branch 'master' into new_YAML_config_branch

asiripanich committed:
Moved the default config file to `inst` folder

asiripanich committed:
Document

asiripanich committed:
Styler

asiripanich commented:
@allenmichael099 at some point, we need to write a documentation for the config file. This can be a [vignette](https://r-pkgs.org/vignettes.html) of the package.

asiripanich committed:
Undocument anon_locations in `run_app()`

asiripanich committed:
- Add `save_config_file()` for saving the default app config file.

asiripanich committed:
Update NEWS

asiripanich committed:
Bump version

asiripanich committed:
Describe params

asiripanich committed:
- Added version number to the dashboard UI
